examples/classic_controllers/classic_controllers.py

```diff
@@ -704,41 +902,507 @@ def control(self, state, reference):
+
+    def control(self, state, torque):
+
+        psi_idx_ = self.get_t_idx_mtpc(torque)
+        psi_opt = self.mtpc[psi_idx_, 3]
+        psi_max_ = self.modulation_control(state)
+        psi_max = min(psi_opt, psi_max_)
+
+        psi_max_idx = self.get_psi_idx_mtpf(psi_max)
+        t_max = np.abs(self.mtpf[psi_max_idx, 1])
+        if np.abs(torque) > t_max:
+            torque = np.sign(torque) * t_max
+
+        if self.torque_control == 'online':
+            i_d, i_q = self.get_i_d_q(torque, psi_max, psi_idx_)
+        else:
+            t_idx = self.get_t_idx(torque)
+            psi_idx = self.get_psi_idx(psi_max)
+
+            if self.i_d_inter[t_idx, psi_idx] <= self.mtpf[psi_max_idx, 2]:
+                i_d = self.mtpf[psi_max_idx, 2]
+                i_q = np.sign(torque) * np.abs(self.mtpf[psi_max_idx, 3])
+                torque = np.sign(torque) * t_max
+            else:
+                i_d = self.i_d_inter[t_idx, psi_idx]
+                i_q = self.i_q_inter[t_idx, psi_idx]
+                if i_d > self.mtpc[psi_idx_, 1]:
+                    i_d = self.mtpc[psi_idx_, 1]
+                    i_q = np.sign(torque) * np.abs(self.mtpc[psi_idx_, 2])
+        if i_d < self.mtpf[psi_max_idx, 2]:
+            i_d = self.mtpf[psi_max_idx, 2]
+            i_q = np.sign(torque) * np.abs(self.mtpf[psi_max_idx, 3])
+
+        if self.plot_torque:
+            if self.k == 0:
+                self.intitialize_torque_plot()
+
+            self.k_list.append(self.k / 10000)
+            self.i_d_list.append(i_d)
+            self.i_q_list.append(i_q)
+            self.torque_list.append(torque)
+            self.psi_list.append(psi_max)
+
+            if self.k % self.update_interval == 0:
+                self.psi_plot.scatter(self.torque_list, self.psi_list, c='tab:blue', s=3)
+                self.torque_plot.scatter(self.psi_list, self.torque_list, c='tab:blue', s=3)
+                self.i_d_q_characteristic_.scatter(self.i_d_list, self.i_q_list, c='tab:blue', s=3)
+
+                self.fig_torque.canvas.draw()
+                self.fig_torque.canvas.flush_events()
+                self.k_list = []
+                self.i_d_list = []
+                self.i_q_list = []
+                self.torque_list = []
+                self.psi_list = []
+
+        i_q = np.clip(i_q, -self.nominal_values[self.i_sq_idx], self.nominal_values[self.i_sq_idx]) / self.limit[self.i_sq_idx]
+        i_d = np.clip(i_d, -self.nominal_values[self.i_sd_idx], self.nominal_values[self.i_sd_idx]) / self.limit[self.i_sd_idx]
+
+        self.k += 1
+
+        return i_q, i_d
+
+    def modulation_control(self, state):
+        eps = state[self.epsilon_idx] * self.limit[self.epsilon_idx]
+        u_sd = state[self.u_sd_idx] * self.limit[self.u_sd_idx]
+        u_sq = state[self.u_sq_idx] * self.limit[self.u_sq_idx]
+        u_alpha, u_beta = self.q([u_sd, u_sq], eps)
```

Comment:
@wallscheid, I just wondered, if the transformation to alpha-beta coordinates is required at this point. 
It is only used to calculate the modulation degree "a" with the length of the voltage vector. 

As only circular voltage set points are used here and the hexagon is neglected, I think this is redundant.

Reply:
Yes, I agree. The voltage vector length used in the following can be directly calculated from dq coordinates.

Reply:
I have changed the calculation of a.